### ISSUE TITLE
Tile visibility - AI keeps track of which tiles it knows about

### ISSUE THREAD

**Issue by QuintillusCFC** [C7-Game/Prototype#104]
The AI should keep track of which tiles it knows about.  This will enable several features:

 - Explorer AI will only explore places the AI doesn't know about
 - Settler AI will only settle where the AI has explored... unless the AI has done a poor job exploring and they just go in a random direction and hope for the best.
 - Defensive AI may eventually be able to defend chokepoints/resources only once they are discovered

It's worth noting that while we have some decent indications that the Civ AI cheats in this regard, and simulates exploration, that doesn't mean we also have to make the AI cheat.  And even if it does cheat, it will be easier to simulate exploration with this data.

**Comment by JimOfLeisure** [comment]
In native Civ3 there is a 32-bit flag to track which civs have discovered each tile, and the recent updates by others to QueryCiv3 add them: <https://github.com/C7-Game/Prototype/blob/c78b9804df659b774d310f53b3bced15753a4a2d/QueryCiv3/SavSections/Tile.cs#L46>

I think the `ExploredBy` is the one you want *when reading from Civ3 saves*. We'll need C7-native logic to allow for > 31 civs (plus barbs). Bit 0 is barbs–yes, all of them–bit 1 is the first human player, and so on.

In other words, the AI should use the same mechanism the player does for determining explored/discovered tiles.

Incidentally, I forget the exact details, but the other flags include "visible now" which includes units and maybe towers, and then either culture owner or territory by civ, and I don't recall what the others are offhand. Not entirely sure we need to have those in a C7 native save as they change frequently and are calculable from current game state.

**Comment by QuintillusCFC** [comment]
Closing as for some reason that didn't happen when the PR got merged (in theory it should happen automatically).
